tested/nat_translation.py

```diff
@@ -0,0 +1,215 @@
+
+        current_translations = data.pop("translations", {})
+        for key, value in current_translations.items():
+            assert language in value
+            current_translations[key] = value[language]
+        translations = {**translations, **current_translations}
+
+        return {
+            key: translate_yaml(value, translations, language, env)
+            for key, value in data.items()
+        }
+    elif isinstance(data, list):
+        return [translate_yaml(item, translations, language, env) for item in data]
+    elif isinstance(data, str):
+        try:
+            return env.from_string(data).render(translations)
+        except TemplateSyntaxError:
+            return data
+    return data
+
+
+def wrap_in_braces(value):
+    """
+    This function will provide the ability to still keep the curly bracket around the
+    translated result. Example: {{ key | braces }} => {sleutel} and {{ key }} => sleutel.
+    """
+    return f"{{{value}}}"
+
+
+class TrackingUndefined(Undefined):
+    missing_keys = list()
+
+    def __str__(self):
+        # Store the missing key name
+        TrackingUndefined.missing_keys.append(self._undefined_name)
+        # Return it in Jinja syntax to keep it in the template
+        return f"{{{{ {self._undefined_name} }}}}"
+
+    __repr__ = __str__
+
+
+def create_enviroment() -> Environment:
+    enviroment = Environment(undefined=TrackingUndefined)
+    enviroment.filters["braces"] = wrap_in_braces
+    return enviroment
+
+
+def generate_new_yaml(yaml_path: Path, yaml_string: str, language: str):
+    file_name = yaml_path.name
+    split_name = file_name.split(".")
+    path_to_new_yaml = yaml_path.parent / f"{'.'.join(split_name[:-1])}-{language}.yaml"
+    with open(path_to_new_yaml, "w", encoding="utf-8") as yaml_file:
+        yaml_file.write(yaml_string)
+
+
+def convert_to_yaml(translated_data: Any) -> str:
+    CustomDumper.add_representer(dict, CustomDumper.custom_representer)
+    return yaml.dump(
+        translated_data, Dumper=CustomDumper, allow_unicode=True, sort_keys=False
+    )
+
+
+def parse_yaml(yaml_stream: str) -> Any:
+    """
+    Parse a string or stream to YAML.
+    """
+    loader: type[yaml.Loader] = cast(type[yaml.Loader], yaml.SafeLoader)
+    yaml.add_multi_constructor("", construct_custom, loader)
+
+    try:
+        return yaml.load(yaml_stream, loader)
+    except yaml.MarkedYAMLError as exc:
+        raise_yaml_error(yaml_stream, exc)
+
+
+def run_translation_with_str(
+    yaml_stream: str, language: str, to_file: bool = True, path: Path | None = None
+) -> tuple[str, list]:
+    parsed_yaml = parse_yaml(yaml_stream)
+    validate_pre_dsl(parsed_yaml)
+
+    enviroment = create_enviroment()
+    translated_data = translate_yaml(parsed_yaml, {}, language, enviroment)
+
+    missing_keys = TrackingUndefined.missing_keys
+    translated_yaml_string = convert_to_yaml(translated_data)
+    if to_file and path is not None:
```

Comment:
I would move this to `run_translation`, which takes the file path as input. And have this method always return a string